test/EndToEnd/WorkloadTests.cs

```diff
@@ -0,0 +1,134 @@
+
+            new DotnetCommand(env.dotnetUnderTest)
+                 .WithEnvironmentVariable("PATH", env.pathValue)
+                 .WithWorkingDirectory(Path.GetDirectoryName(env.projectFile))
+                 .ExecuteWithCapturedOutput(publishArgs)
+                 .Should().Fail()
+                 .And.HaveStdOutContaining("workload packs were not installed")
+                 .And.HaveStdOutContaining("Microsoft.NETcore.App.Runtime.AOT.Cross.browser-wasm");
+        }
+
+        [Fact]
+        public void ItCanPublishBlazorWasmWithWorkloads_NoAOT() => TestBlazorWasmWithWithWorkload();
+
+        [Fact]
+        public void ItCanPublishBlazorWasmWithWorkloads_WithAOT() => TestBlazorWasmWithWithWorkload("/p:RunAOTCompilation=true");
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("/p:RunAOTCompilation=true")]
+        public void ItCanPublishBlazorWasmWithWorkloads_Invariant(string args) => TestBlazorWasmWithWithWorkload($"/p:InvariantGlobalization=true {args}");
+
+        private static void TestBlazorWasmWithWithWorkload(string extraBuildArgs="", [CallerMemberName] string callerName = "")
+        {
+            DirectoryInfo testDirectory = TestAssets.CreateTestDirectory(callingMethod: callerName);
+            WorkloadTestEnvironment env = PrepareTestEnvironment(testDirectory.FullName);
+
+            new DotnetWorkloadCommand(env.dotnetUnderTest)
+                .ExecuteWithCapturedOutput("install microsoft-net-sdk-blazorwebassembly-aot --skip-manifest-update")
+                .Should().Pass();
+
+            string publishArgs = $"publish {extraBuildArgs}";
+            new DotnetCommand(env.dotnetUnderTest)
+                 .WithEnvironmentVariable("PATH", env.pathValue)
+                 .WithWorkingDirectory(Path.GetDirectoryName(env.projectFile))
+                 .Execute(publishArgs)
+                 .Should().Pass();
+        }
+
+        private static WorkloadTestEnvironment PrepareTestEnvironment(string baseDirectory)
+        {
+            string origDotnetPath = Path.GetDirectoryName(RepoDirectoriesProvider.DotnetUnderTest);
+
+            string packsDir = Path.Combine(origDotnetPath, "packs", "Microsoft.NET.Runtime.WebAssembly.Sdk");
+            Directory.Exists(packsDir)
+                .Should().BeFalse($"{packsDir} should not exist. Shared 'dotnet' installation in artifacts, should not have any packs installed");
+
+            string dotnetDirectory = Path.Combine(baseDirectory, "dotnet");
+            DirectoryCopy(origDotnetPath, dotnetDirectory);
+
+            string dotnetUnderTest = Path.Combine(dotnetDirectory, "dotnet");
+            dotnetUnderTest += RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".exe" : "";
+
+            string projectDirectory = Path.Combine(baseDirectory, "project");
+            string projectFile = TestTemplateCreate("blazorwasm", projectDirectory);
+
+            // override PATH since razor rely on PATH to find dotnet
```

Comment:
This should no longer be true. You could try removing this

Reply:
This actually resulted in a cleaner environment, and exposed a bug where emcc couldn't find `python` in the emsdk's python dir, and would fallback on a system python. But because the PATH here didn't include that, it would fail!

So, maybe we should override `PATH` anyway?

Reply:
Sure if it helps the test. Do you mind updating the comment to reflect this?